corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -35,21 +39,50 @@ hqDefine("geospatial/js/geospatial_map", [
                     trash: true
                 },
             });
-            map.addControl(draw);
 
-            map.on('draw.create', updateArea);
-            map.on('draw.delete', updateArea);
-            map.on('draw.update', updateArea);
+            map.addControl(draw);
 
             function getCoordinates(event) {
                 return event.lngLat;
             };
 
-            function updateArea(e) {
-                const data = draw.getAll();
-                const area = turf.area(data);
-                // Restrict the area to 2 decimal points.
-                const rounded_area = Math.round(area * 100) / 100;
+            map.on("draw.update", function(e) {
+                var selectedFeature = e.features[0];
+                if (selectedFeature.geometry.type == 'Polygon') {
+                    // Filter for selected cases when dragging a polygon
+                    var polygon = selectedFeature.geometry;
+                    filterCasesInPolygon(polygon);
+                }
+            });
+
+            map.on('draw.selectionchange', function(e) {
+                // See https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md#drawselectionchange
+                var selectedFeatures = e.features;
+
+                // Check if any features are selected
+                if (selectedFeatures.length > 0) {
```

Comment:
Nit: You don't have to use a comparison operator since any length above 0 will be 'truthy'. This can be rewritten as `if (selectedFeatures.length)`.